nano/lib/blocks.cpp

```diff
@@ -151,8 +152,10 @@ bool nano::block::has_sideband () const
 
 nano::account const & nano::block::representative () const
 {
-	static nano::account rep{ 0 };
-	return rep;
+	/* return nano::hardened_constants::get ().not_an_account; */
```

Comment:
Oh, good catch. If I remember right, I put that comment there because it seemed to me like a good place to use the sentinel, although for some reason I could not find the right places to use it too "on the other side of the == operation". In like, if I changed it there, tests were failing and I couldn't tell what was missing.

Maybe the sentinel isn't fit for this specific site... @clemahieu do you have any idea?

Reply:
I'm not sure what would be causing it without digging in but we don't need to fix up all call sites in one go. I'd say remove the comments also if it's not the implementation we're using.

Reply:
We can remove the comments and replace them with a improvement github issue.